semver.js

```diff
@@ -793,6 +793,56 @@ function toComparators(range, loose) {
   });
 }
 
+exports.comparatorsIntersect = comparatorsIntersect;
+function comparatorsIntersect(compA, compB, loose, platform) {
+  compA = new Comparator(compA, loose);
+  compB = new Comparator(compB, loose);
+
+  if (compA.operator === '') {
+    var rangeB = new Range(compB.value, loose, platform);
+    return satisfies(compA.value, rangeB, loose, platform);
+  } else if (compB.operator === '') {
+    var rangeA = new Range(compA.value, loose, platform);
+    return satisfies(compB.semver, rangeA, loose, platform);
+  }
+  // Same direction increasing
+  return ((compA.operator === '>=' || compA.operator === '>') && (compB.operator === '>=' || compB.operator === '>')) ||
+    // Same direction decreasing
+    ((compA.operator === '<=' || compA.operator === '<') && (compB.operator === '<=' || compB.operator === '<')) ||
+    // Different directions, same semver and inclusive operator
+    (compA.semver.raw === compB.semver.raw &&
+    (compA.operator === '>=' || compA.operator === '<=') && (compB.operator === '>=' || compB.operator === '<=')) ||
+    // Opposite matching directions
+    (cmp(compA.semver, '<', compB.semver, loose) &&
+    ((compA.operator === '>=' || compA.operator === '>') && (compB.operator === '<=' || compB.operator === '<'))) ||
+    (cmp(compA.semver, '>', compB.semver, loose) &&
+    ((compA.operator === '<=' || compA.operator === '<') && (compB.operator === '>=' || compB.operator === '>')));
+}
+
+exports.comparatorSatisfiesRange = comparatorSatisfiesRange;
```

Comment:
Similarly; if this was a prototype method on Comparator it could just be `satisfiesRange`?